pkg/virt-controller/watch/migration.go

```diff
@@ -479,55 +477,59 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 		return fmt.Errorf("vmi is inelgible for migration because another migration job is running")
 	}
 
-	switch migration.Status.Phase {
-	case virtv1.MigrationPending:
+	podNotExistErrFunc := func(podExists bool) error {
```

Comment:
I do not understand how the name expresses what this massive function does.
I would also argue that this should not be a lambda, but a regular method or function that does something meaningful.

It seems that it eventually creates a target pod, but it reaches that after a lot of checks... perhaps you can split each check to a function so we can really follow what this does from some safe distance (i.e. not get polluted into details).

Reply:
This code is wrapped in a lambda, for my understanding, only to defer the unlocking of migrationStartLock which makes a lot of sense (unlocking in each return statement is both error-prone and duplicating code).
If the function's body is only executed once is a very specific context I'm not sure it deserves a new non-lambda function definition.
I moved the function out of the switch-case scope so SonarCloud would not consider this a code-smell, but in second look it's not appropriate - it's more readable and straight forward to place the function body inside the "case" scope as it was before my changes. I will place it back there as it was and will push again.

Reply:
> only to defer the unlocking of migrationStartLock

That makes it a block that should have a meaning, if it does not have a meaning, then something is wrong.

> If the function's body is only executed once is a very specific context I'm not sure it deserves a new non-lambda function definition.

Functions do not exist only in case they need to be called more than once. They exist to abstract pieces of code so it will be easy to read at different levels of details. Modern compilers will inline it if it can, this should not worry the author unless there is a good reason for it.

> I moved the function out of the switch-case scope so SonarCloud would not consider this a code-smell
> I will place it back there as it was and will push again.

You may revert this change, but the warning is correct. Placing so much code in one function or in one case is really bad.
We should be able to read what a function does at some level of abstraction without getting distracted by all small details. Details should be extracted into functions.
I guess it is fine not to handle all the problems in one PR, but you can consider changing this as well in the future.